vapid.go

```diff
@@ -97,3 +97,14 @@ func getVAPIDAuthorizationHeader(
 		vapidPublicKey,
```

Comment:
we also decode and reencode the vapid public key to ensure that it's in `RawURLEncoding` format and not in the padded `URLEncoding` format - we were getting push failures on Chrome otherise if the public key was in `URLEncoding` format. It's kind of unfortunate since it's extra work in the case where it's already RawUrl encoded, but it's a reasonably short string and the application is mostly network-bound and not CPU-bound anyway so we just went with it.

```
decodedVapidPublicKey, err := decodeVapidKey(vapidPublicKey)
if err != nil {
	return "", err
}
return fmt.Sprintf(
	"vapid t=%s, k=%s",
	jwtString,
	base64.RawURLEncoding.EncodeToString(decodedVapidPublicKey),
), nil
```

Reply:
Thanks for the suggestion again, I added decoding/encoding to the public key as well.